# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLLegendElement.h
# Commit: 6141de941e71
# Full Hash: 6141de941e717cdb5485e1aa1ca00ffcd74f8a81
# Author: Mats Palmgren <mats@mozilla.com>
# Date: 2021-01-30 19:41:15
# Regressor Bug: 1683748
# File Overlap Count: 3
# Description:
#   Bug 1683748 - Support Grid/Flex/Table/Column layout for the rendered legend of a fieldset.  r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101555
# ==============================================================================

diff -r 3cf752580f1b -r 6141de941e71 dom/html/HTMLLegendElement.h
--- a/dom/html/HTMLLegendElement.h	Fri Jan 29 19:20:24 2021 +0000
+++ b/dom/html/HTMLLegendElement.h	Sat Jan 30 13:47:10 2021 +0000
@@ -57,6 +57,16 @@
     InlineStart,
     InlineEnd,
   };
+
+  /**
+   * Return the align value to use for the given fieldset writing-mode.
+   * (This method resolves Left/Right to the appropriate InlineStart/InlineEnd).
+   * @param aCBWM the fieldset writing-mode
+   * @note we only parse left/right/center, so this method returns Center,
+   * InlineStart or InlineEnd.
+   */
+  LegendAlignValue LogicalAlign(mozilla::WritingMode aCBWM) const;
+
   /**
    * WebIDL Interface
    */